diff --git a/sys/debian.sh b/sys/debian.sh
index 3613bce8b..78eee9973 100755
--- a/sys/debian.sh
+++ b/sys/debian.sh
@@ -1,8 +1,10 @@
 #!/bin/sh
 # run this from a debian system, docker is fine :)
-#sys/build.sh
 
-# inspired by ios-sdk.sh
+if [ -z "${ARCH}" ]; then
+  ARCH=`uname -m`
+fi
+
 echo "[debian] preparing radare2 package..."
 PKGDIR=sys/debian/radare2/root
 DEVDIR=sys/debian/radare2-dev/root
@@ -14,12 +16,12 @@ mkdir -p "${DEVDIR}/usr/lib"
 mv "${PKGDIR}/usr/lib/"lib*a "${DEVDIR}/usr/lib"
 mv "${PKGDIR}/usr/lib/pkgconfig" "${DEVDIR}/usr/lib"
 for a in ${PKGDIR}/usr/bin/* ; do
-   echo "[debian] strip $a"
-   strip -s "$a" || strip "$a"
+  echo "[debian] strip $a"
+  strip -s "$a" 2> /dev/null || strip "$a" 2>/dev/null
 done
 
 echo "[debian] building radare2 package..."
-make -C sys/debian/radare2
+make -C sys/debian/radare2 ARCH=${ARCH}
 
 echo "[debian] building radare2-dev package..."
-make -C sys/debian/radare2-dev
+make -C sys/debian/radare2-dev ARCH=${ARCH}
